src/nnvm/low_precision_pass.cc

```diff
@@ -0,0 +1,265 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2016 by Contributors
+ * \file low_precision_pass.cc
+ * \brief Return new graph with amp_cast and amp_multicast operators added wherever required
+ */
+
+#include <nnvm/node.h>
```

Comment:
please don't use diamonds for non system includes.

Reply:
this has been used at many places in the codebase, for 3rd party libraries. The link also explains the difference but doesnt explicitly say what is the best practice. In MXNet we use #include <> for all system headers, 3rd party headers etc. Maybe I am missing something, can you elaborate  or provide more links here.

Reply:
it's not used correctly, and it's in the process of being fixed.  If you use diamonds it adds to the search path the standard include folders (the ones from the standard library provided by the compiler). Using quotes just searches in the include path of the project as specified in the build.

If you were to do make install for example of another mxnet version and have nnvm headers in /usr/include/nnvm you would be including these wrong headers first instead of the project (your own). Hence this is a bad practice that should be corrected, that's why I suggested not to do it when introducing new files, as the older ones need to be refactored accordingly.

Reply:
I think that would be expected behavior. First you would provide -I for the path for CFLAGS, which would be searched before sysdirs, and if you intentionally left that out, then you know what you are doing and got what was expected.  Why is it a bad practice ?

Reply:
The reason I asked "Why it is a bad practice?" because you called it not a good practice, and bad practice twice, with no justification for the same. 

The links that you provide talk nothing about "bad practice" or "not a good practice", only what they mean.

I think you missed reading the article you sent.

> There are a number of command-line options you can use to add additional directories to the search path. The most commonly-used option is -Idir, which causes dir to be searched after the current directory (for the quote form of the directive) and ahead of the standard system directories. You can specify multiple -I options on the command line, in which case the directories are searched in left-to-right order.


> I tried to explain above without success, please refer to the compiler documentation:
>  If you need more clarification let's talk in person as seems the converstation is stuck.

I don't like your tone. It was certainly not warranted here. Its an easy way to drive away new contributors.

Reply:
I think this is a misunderstanding, I wanted to talk about this f2f to clarify the order of includes and not discuss further on this PR.  My apologies if my comment didn't come across in a nice way.


 
I was referring to this part of the documentation I sent:

> include <file>
   This variant is used for system header files


> #include "file"
This variant is used for header files of your own program. 

My understanding was that if you do "make install" and have nnvm includes in the system header, it would be used first before the headers of the own program. But according to the doc seems that since we are using -I this won't be the case. 

I don't think there's a need to get angry about this, it's better to learn together.